lib/ansible/plugins/connection/network_cli.py

```diff
@@ -319,7 +319,9 @@ def _connect(self):
         '''
         Connects to the remote device and starts the terminal
         '''
-        if not self.connected:
+        if self.connected is not True:
+            # Replace with truthy-but-not-True value to avoid infinite recursion.
+            self._connected = "connecting"
```

Comment:
I am not clear as to why setting the value of self._connected is required here.

Reply:
The problem is that the second half of `_connect()` calls some cliconf methods which talk to the device, and `send()` is now gated by `ensure_connect` which will... call `_connect()` if `self._connected` is falsy.

I've moved setting `self._connect` to after the connection is established but before the terminal is configured to avoid the infinite recursion. I'd also prefer something to ensure that the terminal is set up before calling other things just in case, but this should work for now.